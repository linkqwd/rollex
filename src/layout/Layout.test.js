import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';

import { Layout } from './index';


describe('Test Layout component', () => {
    it('should render Layout', () => {
        const testText = 'Test text';
        const wrapper = mount(
            <Layout>
                <Layout.Header>{testText}</Layout.Header>
                <Layout.Footer>{testText}</Layout.Footer>
            </Layout>
        );

        expect(wrapper.exists('div')).toEqual(true);
        expect(wrapper.find('header').text()).toEqual(testText);
        expect(wrapper.find('footer').text()).toEqual(testText);
    });

    it('should render Layout with passed props', () => {
        const wrapper = mount(
            <Layout>
                <Layout.Header height='100px' shadow='none'>Hello</Layout.Header>
            </Layout>
        );

        expect(wrapper.exists('div')).toEqual(true);
        expect(wrapper.find('header').props().height).toEqual('100px');
        expect(wrapper.find('header').children().props().shadow).toEqual('none');
    });

    it('renders correctly', () => {
        const rendered = create(
            <Layout>
                <Layout.Header></Layout.Header>
                <Layout.Body></Layout.Body>
                <Layout.Footer></Layout.Footer>
            </Layout>
        ).toJSON();
        expect(rendered).toMatchSnapshot();
    });
});
