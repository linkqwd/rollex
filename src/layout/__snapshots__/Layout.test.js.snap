// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Layout component renders correctly 1`] = `
<div
  className="Layout__MainWrapper-sc-686egp-0 fPpZvL"
>
  <header
    className="Layout__HeaderWrapper-sc-686egp-3 hoXhQk"
  >
    <div
      className="Layout__FixedContainer-sc-686egp-4 eqeLmZ"
    />
  </header>
  <div
    className="Layout__Body-sc-686egp-1 hZRUbq"
  />
  <footer
    className="Layout__Footer-sc-686egp-2 dmwGAt"
  />
</div>
`;
